app/models/stat/anime_genre_breakdown.rb

```diff
@@ -0,0 +1,47 @@
+class Stat
+  class AnimeGenreBreakdown < Stat
+    # fully regenerate data
+    def recalculate!
+      genres = user.library_entries.eager_load(anime: :genres)
+                   .where.not(genres: { slug: nil })
+                   .group(:'genres.slug').count
+
+      self.stats_data = genres
+      stats_data['total'] = genres.values.reduce(:+)
+
+      if new_record?
+        save
+      else
+        update_attribute(:stats_data, stats_data)
+      end
+    end
+
+    def self.increment_genres(user, genres)
```

Comment:
I kind of feel like that these methods can just be set to `increment` and `decrement` and then just used across all the other Stats (so they are all consistent, because we already know it is genres due to `type`)